src/testing_system/internal/usecase/ProcessQuery.py

```
@@ -0,0 +1,37 @@
+                 assistant: IAssistant, 
+                 retriever: IRetriever
+                 ):
+        self.assistant = assistant
+        self.retriever = retriever
+
+    def execute(self, 
+                question:str, 
+                top_k: int = 3, 
+                system_prompt: Optional[str] = None,
+                temperature: Optional[float] = None,
+                metadata: Dict[str,Any] = None
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)